core/server/master/src/main/java/alluxio/master/file/loadmanager/LoadManager.java

```diff
@@ -93,70 +121,130 @@ static class Scheduler {
     private static final long TIMEOUT = 100;
     private final ExecutorService mExecutorService = Executors.newSingleThreadExecutor();
     private final BlockingQueue<Load> mLoadQueue = new LinkedBlockingQueue<>(CAPACITY);
-    private final AtomicInteger mCurrentSize = new AtomicInteger();
+    private final AtomicInteger mCurrentSize = new AtomicInteger(); // scheduler capacity indicator
     private final AtomicLong mIdGenerator = new AtomicLong();
+    private final FileSystemMaster mFileSystemMaster;
+    private final FileSystemContext mFileSystemContext;
+
+    public Scheduler(FileSystemMaster fileSystemMaster, FileSystemContext fileSystemContext) {
+      mFileSystemMaster = fileSystemMaster;
+      mFileSystemContext = fileSystemContext;
+    }
 
     void schedule(Load load)
-            throws ResourceExhaustedException, InterruptedException {
-      if (mCurrentSize.get() == CAPACITY) {
-        throw new ResourceExhaustedException(
-                "Insufficient capacity to enqueue load tasks!");
+            throws InterruptedException {
+      if (mCurrentSize.get() >= CAPACITY) {
+        waitForLoad();
       }
 
       boolean offered = mLoadQueue.offer(load, TIMEOUT, TimeUnit.MILLISECONDS);
       if (offered) {
         mCurrentSize.incrementAndGet();
         mExecutorService.submit(() -> {
-          try {
-            runLoad(load);
-          } catch (AlluxioRuntimeException e) {
-            handleErrorOnStatuses(); // handle based on status
-          } catch (TimeoutException e) {
-            // add retry and handle timeout caused by checking available workers
-          }
+          runLoad(load);
         });
       } else {
-        LOG.warn("Cannot enqueue load to the queue, may lose track on this load!"
+        LOG.warn("Cannot enqueue load to the queue, may lose track on this load task!"
```

Comment:
Didn't pay attention to this. It's probably a bad idea to lose track of a load job and only leave a warning in the log. This should fail and give user an error message (throw exception here).